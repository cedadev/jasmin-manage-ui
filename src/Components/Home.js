import React from 'react';

import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Button from 'react-bootstrap/Button';

import { LinkContainer } from 'react-router-bootstrap';


const Home = () => (
    <Row>
        <Col>
            <div class="jumbotron">
                <h1>JASMIN Projects Portal</h1>
                <p>Welcome to the JASMIN Projects Portal.</p>
                <p>Using this portal, you can request JASMIN resources for your project and track their status.</p>
                <LinkContainer to="/projects">
                    <Button variant="primary" size="lg">Manage my projects</Button>
                </LinkContainer>
            </div>
        </Col>
    </Row>
);

export default Home;
