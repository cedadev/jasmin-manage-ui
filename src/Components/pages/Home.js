import React from 'react';

import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Jumbotron from 'react-bootstrap/Jumbotron';


const Home = () => (
    <Row>
        <Col>
            <Jumbotron>
                <h1>JASMIN Projects Portal</h1>
                <p>Welcome to the JASMIN Projects Portal.</p>
                <p>Using this portal, you can request JASMIN resources for your project and track their status.</p>
            </Jumbotron>
        </Col>
    </Row>
);

export default Home;
