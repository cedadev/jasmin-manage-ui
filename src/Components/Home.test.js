import React from 'react';
// import { render, screen } from '@testing-library/react';
// import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Home from './Home';


it('renders the JASMIN Projects Portal Homepage correctly', () => {
    const component = renderer.create(
        <MemoryRouter>
            <Home />
        </MemoryRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});



// test('renders the JASMIN Projects Portal Homepage correctly', () => {
//     // Render the Home component within a Router
//     render(
//         <Router>
//             <Home />
//         </Router>
//     );

//     // Check for the presence of specific elements and text
//     expect(screen.getByText('JASMIN Projects Portal')).toBeInTheDocument();
//     expect(screen.getByText('Welcome to the JASMIN Projects Portal.')).toBeInTheDocument();
//     expect(screen.getByText('Using this portal, you can request JASMIN resources for your project and track their status.')).toBeInTheDocument();
//     expect(screen.getByText('Manage my projects')).toBeInTheDocument();
//     expect(screen.getByRole('button', { name: 'Manage my projects' })).toBeInTheDocument();
// });