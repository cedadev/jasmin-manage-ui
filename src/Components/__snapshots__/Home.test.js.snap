// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the JASMIN Projects Portal Homepage correctly 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      className="jumbotron"
    >
      <h1>
        JASMIN Projects Portal
      </h1>
      <p>
        Welcome to the JASMIN Projects Portal.
      </p>
      <p>
        Using this portal, you can request JASMIN resources for your project and track their status.
      </p>
      <a
        className="btn btn-primary btn-lg"
        href="/projects"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Manage my projects
      </a>
    </div>
  </div>
</div>
`;
